src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -115,9 +116,11 @@ public override TypeSyntax VisitDynamicType(IDynamicTypeSymbol symbol)
             public override TypeSyntax VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
             {
                 // TODO(https://github.com/dotnet/roslyn/issues/39865): generate the calling convention once exposed through the API
-                var parameters = symbol.Signature.Parameters.Select(p => p.Type)
-                    .Concat(SpecializedCollections.SingletonEnumerable(symbol.Signature.ReturnType))
-                    .SelectAsArray(t => SyntaxFactory.Parameter(SyntaxFactory.MissingToken(SyntaxKind.IdentifierToken)).WithType(t.GenerateTypeSyntax()));
+                var parameters = symbol.Signature.Parameters.Select(p => (p.Type, RefKindModifiers: CSharpSyntaxGenerator.GetParameterModifiers(p.RefKind)))
+                    .Concat(SpecializedCollections.SingletonEnumerable((
+                        Type: symbol.Signature.ReturnType,
+                        RefKindModifiers: CSharpSyntaxGenerator.GetParameterModifiers(symbol.Signature.RefKind, forFunctionPointerReturnParameter: true))))
+                    .SelectAsArray(t => SyntaxFactory.Parameter(SyntaxFactory.MissingToken(SyntaxKind.IdentifierToken)).WithModifiers(t.RefKindModifiers).WithType(t.Type.GenerateTypeSyntax()));
 
                 return AddInformationTo(
                     SyntaxFactory.FunctionPointerType(SyntaxFactory.SeparatedList(parameters)), symbol);
```

Comment:
how do you specify the `cdecl` bit here?

Reply:
You can't today. That's why there's a comment above about it.